kernel/syscall.c

```diff
@@ -3145,6 +3155,7 @@ static long _syscall(void* args_)
         }
         case SYS_myst_add_symbol_file:
         {
+#ifndef MYST_RELEASE
```

Comment:
Just realized that a developer might still want to use myst-gdb to debug the app code inside Mystikos, when using the prebuilt Mystikos binaries (Mystikos "release build"). Should the switch for debug symbol related APIs be based on the debug flag in config.json, instead of the compiling switch? 

Alternatively, we can have two sets of binaries of Mystikos in a release, one for "release build" app, where all debug capability is disabled, one for "debug build" app, where the Mystikos binaries themselves are "debug build".

Reply:
Oh good point. Can we use the SGX debug attribute instead? We fetch it on enclave startup like this:

```C++
static int _test_oe_debug_mode(void)
{
    int ret = -1;
    uint8_t* buf = NULL;
    size_t buf_size;
    oe_report_t report;

    if (oe_get_report_v2(0, NULL, 0, NULL, 0, &buf, &buf_size) != OE_OK)
        goto done;

    if (oe_parse_report(buf, buf_size, &report) != OE_OK)
        goto done;

    if (!(report.identity.attributes & OE_REPORT_ATTRIBUTES_DEBUG))
        goto done;

    ret = 0;

done:

    if (buf)
        oe_free_report(buf);

    return ret;
}
```

Reply:
Yes, SGX debug attribute decides whether the symbols are accessible by GDB.

Reply:
So is it an acceptable solution?

Reply:
For developer app debugging support in this PR, the solution is OK. We can discuss the "two sets of binaries" idea in a different thread.

Reply:
Okay, I will allow debugging if the attestation of the SGX properties passes. Also I will add a ``-no-debug-symbols`` option to we can selectively speed up tests, solutions, and demos. :)

Reply:
Finished. We enable debug symbols when two conditions are satisfied:
- The SGX debug attribute is on
- ``/proc/self/status:TracerPid`` is non-zero